galaxy_ng/app/access_control/fields.py

```diff
@@ -0,0 +1,56 @@
+from django.contrib.auth.models import Permission
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from galaxy_ng.app.models import auth as auth_models
+
+
+class GroupPermissionField(serializers.Field):
+    def to_representation(self, value):
+        rep = []
+        for group in value:
+            rep.append({
+                'id': group.id,
+                'name': group.name,
+                'object_permissions': value[group]
+            })
+        return rep
+
+    def to_internal_value(self, data):
+        internal = {}
+        for group_data in data:
+            perms = group_data['object_permissions']
+
+            # validate that the permissions exist
+            for perm in perms:
+                permission_components = perm.split('.')
+                filter = {}
+                if len(permission_components) == 2:
+                    filter = {
+                        "content_type__app_label": permission_components[0],
+                        "codename": permission_components[1]
+                    }
+                else:
+                    filter = {
+                        "codename": permission_components[0]
+                    }
+
+                if not Permission.objects.filter(**filter).exists():
```

Comment:
This code will make a N implicit SQL queries, for N of groups.

Reply:
is there a better way to check that a list of permissions exist?

Reply:
Not sure how to express it in Django ORM, normally in PostgreSQL you could use row_construct + IN clause or JOIN with a subquery.